octoprint_mrbeam/static/js/convert.js

```diff
@@ -252,6 +246,9 @@ $(function () {
         self.save_material_settings = function () {
             var name = self.save_custom_material_name();
             var key = self._replace_non_ascii(name).toLowerCase();
+            if(self.incompatible_custom_materials_keys().includes(key)){
+                key = self._replace_non_ascii(MRBEAM_LASER_HEAD_SERIAL).toLowerCase() + " " + key;
+            }
```

Comment:
So if I make a custom material on top of a custom material etc.... will it be saved as 
`laserhead S laserhead S laserhead S my custom material` ?

Reply:
So first thing we prepend via the serial number and not the model. Secondly, this only happens when you save a material with an incompatible material name. Upon saving again with the same name, it won't get prepended again but rather it would add the new custom materials to the already saved key: serial + name.

There is a corner case where this can happen if the user flipped the laserheads again and tried to save the incompatible custom settings while the compatible ones has the same name then this will happen and a new key with the SERIAL + SERIAL + name will be saved => This was just resolved